src/errorpage.cc

```diff
@@ -810,67 +812,60 @@ ErrorState::~ErrorState()
 int
 ErrorState::Dump(MemBuf * mb)
 {
-    MemBuf str;
+    PackableStream out(*mb);
+    const auto &encoding = CharacterSet::RFC3986_UNRESERVED();
+
+    out << "?subject=" <<
+        AnyP::Uri::Encode(SBuf("CacheErrorInfo - "),encoding) <<
+        AnyP::Uri::Encode(SBuf(errorPageName(type)), encoding);
+
+    SBufStream body;
+    body << "CacheHost: " << getMyHostname() << "\r\n" <<
+        "ErrPage: " << errorPageName(type) << "\r\n" <<
+        "TimeStamp: " << Time::FormatRfc1123(squid_curtime) << "\r\n" <<
+        "\r\n";
+
     char ntoabuf[MAX_IPSTRLEN];
+    body << "ClientIP: " << src_addr.toStr(ntoabuf, MAX_IPSTRLEN) << "\r\n";
```

Comment:
Since we are polishing this output for other reasons, if it does not result in information loss, I would use the std::ostream output operator for printing Ip::Address objects in this function.

Reply:
TBH I found it contradictory, but after reviewing the (IMO misnamed) toUrl implementation, I think it can fit here.
Changing